jams/jamongo.py

```diff
@@ -0,0 +1,171 @@
+
+        Parameters
+        ----------
+        annotations_list : jams.AnnotationArray
+
+        Returns
+        -------
+        converted_annotations : list
+        """
+        return []
+
+
+class JamsMongo(object):
+    """Model class for interfacing with JAMS Mongo Collection."""
+
+    def __init__(self, connection_str=None,
+                 client=None,
+                 database_name="jams"):
+        """
+        Pass the parameters necessary to create a MongoClient connection to
+        mongodb.
+
+        If no parameters are provided, just tries to connect to your
+        local mongod.
+
+        Parameters
+        ----------
+        connection_str : str or None
+            Mongo connection string.
+
+        client : pymongo.mongo_client.MongoClient
+            A MongoClient reference. Use this if you want to pass the client
+            instead of having JamsMongo create the client for you from the
+            connection_str.
+
+            (This is useful for unit tests / mongomock.)
+
+        database_name : str
+            The name of the database to use on your client.
+        """
+        self.connection_str = connection_str
+        self.client = client
+        self.database_name = database_name
+        self._db = None
+
+    def __enter__(self):
+        """Set up your mongo collection, and return the object to access it."""
+        self._connect_to_mongo()
+        return self
+
+    def __exit__(self, type, value, traceback):
+        pass
+
+    @property
+    def db(self):
+        return self._db
+
+    @property
+    def audio(self):
+        """Return a pointer to the audio collection."""
+        return self._db['audio']
+
+    @property
+    def annotations(self):
+        """Return a pointer to the annotations collection."""
+        return self._db['annotations']
+
+    def _connect_to_mongo(self):
+        """Connect to mongo database using the connection string
+        provided in the constructor.
+
+        Sets self._db for access in future
+        """
+        if self.client is None:
+            self.client = pymongo.MongoClient(self.connection_str)
+
+        self._db = self.client[self.database_name]
+
+    def import_jams(self, jams_object):
+        """Import a JAMS object into mongo.
+
+        Puts JAMS metadata into the "Audio" collection,
+        and puts the Annotations into the "Annotations" collection.
+
+        Parameters
+        ----------
+        jams_object : jams.JAMS
+        """
+        # Extract the JAMS metadata & insert it into the Audio collection
+        audio_id = self.insert_jams_metadata(jams_object.file_metadata)
+
+        # Extract the JAMS Annotations and prepare them for mongification.
+        annotation_list = convert_annotation_list(jams_object.annotations)
+
+        # insert them into the Annotations collection.
+        annotation_ids = self.insert_annotations(annotation_list, audio_id)
```

Comment:
Maybe simpler to have a method that inserts one at a time, and then iterate over the annotation_list?